### PR TITLE
Most connected profiles

### PR DESCRIPTION
**Pull request by talos**
* Class method to find most connected profiles
* Tests
* Surfaceing use of endpoint in `activity.html`

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -312,6 +313,23 @@ def get_most_complete_profiles(cls, limit=10):
...
+        '''
+        return User.query_in_deployment().\
+                add_column(func.count(func.distinct(
+                    cast(Email.to_user_id, String) + '-' +
+                    cast(Email.from_user_id, String)))).\
+                filter((User.id == Email.from_user_id) | (User.id == Email.to_user_id)).\
+                group_by(User).\
+                order_by(func.count(func.distinct(
+                    cast(Email.to_user_id, String) + '-' +
+                    cast(Email.from_user_id, String))).desc()).\
```
Comment (toolness) [inline app/models.py]: purely from a readability standpoint, do you think it'd be useful to set

```python
count_of_unique_emails = func.count(func.distinct(cast(Email.to_user_id, String) + '-' +
                                                 cast(Email.from_user_id, String)))
```

and then use that twice in the query? I'm fine either way, but it's just a thought.

File: app/templates/activity.html [hunk 2]
```diff
@@ -74,4 +74,18 @@
...
 </ol>
 
+<h3>{{ gettext('Most Connected') }}</h3>
+
+<ol>
+  {% for user, score in most_connected_profiles %}
+    <li>
+      <p>{{ user_link(user) }}</p>
+      <small>
+        {{ score }} connections
```
Comment (toolness) [inline app/templates/activity.html]: Should this be internationalized now, or should we file an issue to do it later?

**Comment by toolness** [comment]
It looks like the actual tests in the travis CI build passed. I think this looks fine!

File: app/models.py (hunk 1 above)
Comment (talos) [inline app/models.py]: :+1:

File: app/templates/activity.html (hunk 2 above)
Comment (talos) [inline app/templates/activity.html]: Smart.  Dealing with this and missing internationalization above.

**Comment by talos** [comment]
Yep, the build failures seem to have been issues with the image, now resolved.

**Pull request merged**